Fix principal exclusion in case of comma in the name (#50)
strimzi/kafka-quotas-plugin: README.md

client.quota.callback.static.storage.check-interval=5

# Optional list of principals not to be subjected to the quota

> client.quota.callback.static.excluded.principal.name.list=principal1;principal2

Reviewer: As we compere this with the `super.users` option from Kafka -> should this also expect the `User:` prefix to be used?

Author: From my understanding, we are here just checking that the name of the `KafkaPrincipal` inside is in the list of excluded principals.
Kafka (at least what I see here https://github.com/apache/kafka/blob/53ff1a5a589eb0e30a302724fcf5d0c72c823682/clients/src/main/java/org/apache/kafka/common/utils/SecurityUtils.java#L63) splits the whole `User: xxx` by `:` and the `User` part uses as a type of the `KafkaPrincipal` and everything after `:` as name.

Author: So I think that it should not be `User:principal1;User:principal2`, as only the name part is used anyway.

Reviewer: And how do you know that type of principal should it apply to?

Author: So I think that originally the plugin assumed that the principals will be of type `User`.
But I can add parsing similar as Kafka has.
And in case that there will be nothing like `User:` or whatever, it will default to `User` in the check.

Author: Yeah, maybe it's something that we should discuss more.
Based on https://kafka.apache.org/37/javadoc/org/apache/kafka/common/security/auth/KafkaPrincipal.html , the `User` as a type will be used always for the simple authorizer. However based on the authz, there can be something else. So maybe to prevent issues in the future, it would make sense to follow the `super.users` completely.

Reviewer 2: > Yeah, maybe it's something that we should discuss more.
Based on https://kafka.apache.org/37/javadoc/org/apache/kafka/common/security/auth/KafkaPrincipal.html , the User as a type will be used always for the simple authorizer. 

Right.   I did wonder whether it would be possible to allow exclusions to be specified for other principal types  (`Group:` for instance), I came to the conclusion that the `quotaMetricTags` API doesn't allow that possibility. `KafkaPrincipal` is just the user.

I don't think prefixing the principals in the list with User: gets us much.    If we did go that route, we'd have to fail if ever configured with a principal prefix apart from `User:`.

Author: Based on the above findings, I guess we are keeping it as it is?

Reviewer 3: So ... I see that the quota API seems to use always `User` type as stated above. At the same time, we could be prepared for future changes if a different type could be used. What about having the `User:principal1;User:principal2` as the official format but defaulting to `User` when it's absent so allowing even the current `principal1:principal2` ? Maybe it's something suggested by @im-konge previously.

Reviewer 2: I think the Kafka API is unlikely to change any time soon, so I don't think words like "at the moment" really help.  If you want, add an example to the table in the README illustrating thus illustrating that the principal is unprefixed.

Author: Discussed on the community call on May 2nd: We will follow the pattern of `super.users` completely, thus prefixing the principal with `User:` prefix.

I will change the PR accordingly.

| client.quota.callback.static.produce                                      | Double  | 1.7976931348623157e+308 | Produce bandwidth rate quota (in bytes)                                                                                                                                                                     |
| client.quota.callback.static.fetch                                        | Double  | 1.7976931348623157e+308 | Consume bandwidth rate quota (in bytes)                                                                                                                                                                     |
| client.quota.callback.static.request                                      | Double  | 1.7976931348623157e+308 | Request processing time quota (in seconds)                                                                                                                                                                  |

> | client.quota.callback.static.excluded.principal.name.list                 | String  | ""                      | Semicolon-separated list of principals excluded from the quota. Each principal have to be prefixed with `User:` prefix to exclude it from the quota.                                                        |

Reviewer 2: ```suggestion
| client.quota.callback.static.excluded.principal.name.list                 | String  | ""                      | Semicolon-separated list of principals excluded from the quota. Each principal has to be prefixed with `User:`.                                                        |
```
have -> has
"to exclude it from the quota" is repetitious and slightly confusing.

Author: Ah yes, thanks 😄
